#include "addcabinetsdialog.h"
#include "ui_addcabinetsdialog.h"
#include<QMessageBox>
#include <QShortcut>

AddCabinetsDialog::AddCabinetsDialog(QWidget *parent)
    : QDialog(parent)
    , ui(new Ui::AddCabinetsDialog)
    , schedule(std::make_unique<DataSchedule>())
    , parseJson(std::make_unique<ScheduleJsonParser>())
{
        ui->setupUi(this);

     this->setLayout(ui->gridLayout);

        ManagerWorker w;
        w.CreateDir(".","–†–∞—Å–ø–∏—Å–∞–Ω–∏–µ");

        QShortcut* shortcut = new QShortcut(QKeySequence("Enter"),this);
        connect(shortcut, &QShortcut::activated, this, [this](){
            auto success = ui->lineEditCab->text().isEmpty();

            if(!success)
            {
                ui->listWidgetPriview->addItem("–ö–∞–±–∏–Ω–µ—Ç:" + ui->lineEditCab->text());

            }
        });

    connect(ui->btnSave,&QPushButton::clicked,[this](){
        auto success = ui->lineEditCab->text().isEmpty();

        if(!success)
        {
            ui->listWidgetPriview->addItem("–ö–∞–±–∏–Ω–µ—Ç:" + ui->lineEditCab->text());

        }

    });



    connect(ui->listWidgetPriview,&QListWidget::clicked,[this](const QModelIndex& index){
        int curr = index.row();

        QListWidgetItem* item = ui->listWidgetPriview->item(curr);

        if (item) {
            delete ui->listWidgetPriview->takeItem(curr);
        }

    });

    connect(ui->btnHelp,&QPushButton::clicked,[this](){
        QMessageBox::information(
            this,
            "–ü–æ–¥—Å–∫–∞–∑–∫–∞ –ø–æ –º–µ–Ω—é",
            "üìå –ö–∞–∫ —Ä–∞–±–æ—Ç–∞—Ç—å —Å –º–µ–Ω—é:\n\n"
            "1Ô∏è‚É£ –í–≤–µ–¥–∏—Ç–µ –¥–∞–Ω–Ω—ã–µ –∫–∞–±–∏–Ω–µ—Ç–∞ –∏ –Ω–∞–∂–º–∏—Ç–µ ¬´–°–æ—Ö—Ä–∞–Ω–∏—Ç—å¬ª.\n"
            "   –î–∞–Ω–Ω—ã–µ –ø–æ—è–≤—è—Ç—Å—è –≤ —Å–ø–∏—Å–∫–µ.\n\n"
            "2Ô∏è‚É£ –ï—Å–ª–∏ –∫–∞–∫–∏–µ-—Ç–æ –¥–∞–Ω–Ω—ã–µ –≤–≤–µ–¥–µ–Ω—ã –Ω–µ–≤–µ—Ä–Ω–æ ‚Äî\n"
            "   –ø—Ä–æ—Å—Ç–æ –Ω–∞–∂–º–∏—Ç–µ –Ω–∞ —ç—Ç–æ—Ç –∫–∞–±–∏–Ω–µ—Ç –≤ —Å–ø–∏—Å–∫–µ,\n"
            "   –æ–Ω –±—É–¥–µ—Ç —É–¥–∞–ª—ë–Ω.\n\n"
            "3Ô∏è‚É£ –ï—Å–ª–∏ –µ—Å—Ç—å –¥—É–±–ª–∏–∫–∞—Ç—ã ‚Äî –Ω–µ –ø–µ—Ä–µ–∂–∏–≤–∞–π—Ç–µ.\n"
            "   –ü—Ä–∏ –æ—Ç–ø—Ä–∞–≤–∫–µ ARDA –∞–≤—Ç–æ–º–∞—Ç–∏—á–µ—Å–∫–∏ –∏—Ö –æ—Ç—Å–µ–µ—Ç.\n\n"
            "4Ô∏è‚É£ –ö–æ–≥–¥–∞ –≤—Å—ë –ø—Ä–æ–≤–µ—Ä–µ–Ω–æ –∏ –¥–æ–±–∞–≤–ª–µ–Ω–æ,\n"
            "   –Ω–∞–∂–º–∏—Ç–µ –∫–Ω–æ–ø–∫—É ¬´–û—Ç–ø—Ä–∞–≤–∏—Ç—å¬ª."
        );

    });

    connect(ui->btnGo,&QPushButton::clicked,this,&AddCabinetsDialog::SendCabinetsDataToJson);

}

AddCabinetsDialog::~AddCabinetsDialog()
{
    delete ui;
}

bool AddCabinetsDialog::SendCabinetsDataToJson()
{
    if(ui->listWidgetPriview->count() == 0){
        QMessageBox::warning(this,"–û—à–∏–±–∫–∞ –æ—Ç–ø—Ä–∞–≤–∫–∏","–ê –∫–∞–±–∏–Ω–µ—Ç—ã –≥–¥–µ?");
        return false;

    }

    for(int i = 0; i<ui->listWidgetPriview->count(); i++)
    {
        QString str = std::move(ui->listWidgetPriview->item(i)->text().remove("–ö–∞–±–∏–Ω–µ—Ç:"));
        schedule->cabinets.insert(str);
    }
    QString file = "–†–∞—Å–ø–∏—Å–∞–Ω–∏–µ/cabinets.json";

    bool success = parseJson->setDataCabinets(file,schedule->cabinets);
    if(success)
    {
        QMessageBox::warning(this,"–û—à–∏–±–∫–∞ –æ—Ç–ø—Ä–∞–≤–∫–∏","—á—Ç–æ —Ç–æ –≤—Å—Ç–∞–≤–∏—Ç—å –¥–∞–Ω–Ω—ã–µ –Ω–µ –ø–æ–ª—É—á–∏—Å–ª–æ—Å—å\n\n –ü—Ä–æ–≤–µ—Ä—å –ø–∞–ø–∫—É –†–∞—Å–ø–∏—Å–∞–Ω–∏–µ –ø–æ –ø—É—Ç–∏ –ø—Ä–æ–≥—Ä–∞–º–º—ã –∏–ª–∏ –ø–µ—Ä–µ–∑–∞–ø—É—Å—Ç–∏ –ø—Ä–æ–≥—É");

        return success;
    }

    return success;

}
